lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -13,13 +13,7 @@
  * @TODO: Clean up attribute token class after is_true addition
  * @TODO: Prune whitespace when removing classes/attributes: e.g. "a b c" -> "c" not " c"
  * @TODO: Skip over `/` in attributes area, split attribute names by `/`
- * @TODO: Decode HTML references/entities in class names when matching.
- *        E.g. match having class `1<"2` needs to recognize `class="1&lt;&quot;2"`.
- * @TODO: Decode character references in `get_attribute()`
  * @TODO: Properly escape attribute value in `set_attribute()`
```

Comment:
Does that still apply? As long as we don't allow the `"` character, everything should be fine even if nothing else is escaped?

Reply:
I'm not following your question. can you rephrase it?

the `class` value of `a&#x65a` should be interpreted as having class `aaa`, so the stylesheet to apply rules to it would look like this…

```css
.aaa { … }
```

these TODOs thus addressed two different issues:
 - if we search for a tag with class `wp-block` it should find a tag even if it's written in the tag as `class="wp&#x2dblock"` or `class='wp&#45block'` or any other variant with character references. strangely the class `&#x1FFFF` (and others like it with invalid character references) map to the CSS class name `�`.
 - when decoding attribute values PHP has been returning potentially invalid strings since `html_entity_decode` is both broken by implementation and by design.